src/dotnet-install.sh

```diff
@@ -787,13 +807,109 @@ downloadwget() {
     return 0
 }
 
+get_download_link_from_aka_ms() {    
+    eval $invocation
+
+    #detect OS
+    osname="$(get_current_os_name)" || return 1
+    say_verbose "OS to use: '$osname'" 
+
+    #choose the product
+    if [[ "$runtime" == "dotnet" ]]; then
+        product="dotnet-runtime"
+    elif [[ "$runtime" == "aspnetcore" ]]; then
+        product="aspnetcore-runtime"
+    elif [ -z "$runtime" ]; then
+        product="dotnet-sdk"
+    else
+        #impossible combination
+        return 1
+    fi
+    say_verbose "Product to use: '$product'" 
+
+    #quality is not supported for LTS or current channel
+    if [[ ! -z "$quality"  && ("$channel" == "LTS" || "$channel" == "Current") ]]; then
+        quality=""
+        say "Specifying quality for current or LTS channel is not supported, the quality will skipped"
+    fi
+
+    #TODO: check if any other restricions for channel should be applied
+
+    say_verbose "Retrieving primary named payload URL from aka.ms for channel '$channel', quality '$quality' product '$product', os '$osname', architecture '$normalized_architecture'" 
+
+    #construct aka.ms link
+    aka_ms_link="https://aka.ms/dotnet/$channel" 
+    if [[ ! -z "$quality" ]]; then
+        aka_ms_link="$aka_ms_link/$quality"
+    fi
+    aka_ms_link="$aka_ms_link/$product-$osname-$normalized_architecture.tar.gz"
+    say_verbose "Constructed aka.ms link: $aka_ms_link"
+
+    #get HTTP response
+    response="$(get_http_header "$aka_ms_link")"
+
+    say_verbose "Received response: $response"
+    http_code=$( echo "$response" | awk '$1 ~ /^HTTP/ {print $2}' | head -1 )
+
+    #if HTTP code is 301 (Moved Permanently), the redirect link exists
+    if [[ "$http_code" == "301" ]]; then
+        aka_ms_download_link=$( echo "$response" | awk '$1 ~ /^Location/{print $2}' | head -1 )
+        if [[ -z "$aka_ms_download_link" ]]; then
+            say_verbose "The aka.ms link is not valid: failed to get redirect location"
+            return 1
+        fi
+        say_verbose "The redirect location retrieved: $aka_ms_download_link"
+        return 0
```

Comment:
minor: consider to return `$aka_ms_download_link` using `echo` before this line, not relying on rewriting the global variable